src/plots/cartesian/tick_defaults.js

```diff
@@ -29,29 +29,27 @@ module.exports = function handleTickDefaults(containerIn, containerOut, coerce,
         delete containerOut.tickcolor;
     }
 
+    var showAttrDflt = getShowAttrDflt(containerIn);
+
     var showTickLabels = coerce('showticklabels');
     if(showTickLabels) {
         Lib.coerceFont(coerce, 'tickfont', options.font || {});
         coerce('tickangle');
 
-        var showAttrDflt = getShowAttrDflt(containerIn);
-
-        if(axType !== 'category') {
-            var tickFormat = coerce('tickformat');
-            if(!options.noHover) coerce('hoverformat');
-
-            if(!tickFormat && axType !== 'date') {
-                coerce('showexponent', showAttrDflt);
-                coerce('exponentformat');
-            }
-        }
-
         var tickPrefix = coerce('tickprefix');
         if(tickPrefix) coerce('showtickprefix', showAttrDflt);
 
         var tickSuffix = coerce('ticksuffix');
         if(tickSuffix) coerce('showticksuffix', showAttrDflt);
```

Comment:
These are not quite right still.

http://codepen.io/etpinard/pen/MKMNXe

`tickprefix` and `ticksuffix` have an effect (and should) of the hover label; they should be coerced regardless of the `showTickLabels`  value.